libs/plugin-api/src/main/java/org/elasticsearch/plugin/api/settings/IntSetting.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.plugin.api.settings;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation used to mark a setting of type integer
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface IntSetting {
```

Comment:
Any reason we decided to provide max, but not min? Same for long setting.

Reply:
good point, if we have max, we can have min too. In fact I am not using max or min in this PR, as it does not contain validation. Perhaps I remove max in this PR, but include max and min in another that will actually use these attributes?